src/condor_tools/htcondor_cli/annex.py

```diff
@@ -0,0 +1,81 @@
+import os
+
+import htcondor
+
+from htcondor_cli.noun import Noun
+from htcondor_cli.verb import Verb
+
+
+class Shutdown(Verb):
+    """
+    Shut an HPC annex down.
+    """
+
+
+    options = {
+        "annex_name": {
+            "args": ("annex_name",),
+            "help": "annex name",
+        },
+    }
+
+
+    #
+    # This (presently) only shuts down currently-operating annexes, and
+    # only annexes for which every startd has successfully reported to the
+    # collector.  We should add an option to scan the queue for the SLURM
+    # job IDs of every annex request with a matching name and scancel the
+    # it; this shouldn't be the default because it will require the user to
+    # log in again.
+    #
+    def __init__(self, logger, annex_name, **options):
+        annex_collector = htcondor.param.get("ANNEX_COLLECTOR", "htcondor-cm-hpcannex.osgdev.chtc.io")
+        collector = htcondor.Collector(annex_collector)
+        location_ads = collector.query(
+            ad_type=htcondor.AdTypes.Master,
+            constraint=f'AnnexName =?= "{annex_name}"',
+        )
+
+        if len(location_ads) == 0:
+            print(f"No resources found in annex '{annex_name}'.")
+            return
+
+        password_file = htcondor.param.get("ANNEX_PASSWORD_FILE", "~/.condor/annex_password_file")
+        password_file = os.path.expanduser(password_file)
```

Comment:
FYI, the `pathlib` `Path` object (`from pathlib import Path`) is really good for handling paths and also has the `expanduser()` method:

```
>>> password_file = Path("~/.condor/annex_password_file")
>>> password_file = password_file.expanduser()
>>> password_file
PosixPath('/Users/jpatton/.condor/annex_password_file')
```

Reply:
Resolving and noting in JIRA ticket.